src/ape_run/_cli.py

```diff
@@ -114,14 +118,33 @@ def commands(self) -> Dict[str, Union[click.Command, click.Group]]:
         if not self._project.scripts_folder.exists():
             return {}
 
+        return self._get_cli_commands(self._project.scripts_folder)
+        # TODO: loop through and find subdirectories
+        # refactor logic to be recursive function
+        # (couple different methods that call eachother)
+        # in each subdirectory change those to groups
+        # each python module turn it to scripts
+
+    def _get_cli_commands(self, base_path: Path) -> Dict:
         commands = {}
-        for filepath in self._project.scripts_folder.glob("*.py"):
+
+        for filepath in base_path.iterdir():
             if filepath.stem.startswith("_"):
                 continue  # Ignore any "private" files
 
-            cmd = self._get_command(filepath)
-            if cmd:  # NOTE: Don't allow calling commands that failed to load
-                commands[filepath.stem] = cmd
+            elif filepath.is_dir():
+                group = click.Group(name=filepath.stem)
+                subcommands = self._get_cli_commands(filepath)
+                for subcommand in subcommands.values():
+                    group.add_command(subcommand)
+                commands[filepath.stem] = group
+                # TODO: commands is a group, subcommands is a networkboundcommand.
```

Comment:
This is not true, groups are commands (but we already discussed this :) )

- [ ] Delete this TODO comment